File: StanfordSoftware/Samples/StanfordPOSTagger.Fsharp.Samples/TaggerDemo.fs
module TaggerDemo

open java.io
open java.util

open edu.stanford.nlp.ling
open edu.stanford.nlp.tagger.maxent;

let model = @"..\..\..\..\temp\stanford-postagger-2013-06-20\models\wsj-0-18-bidirectional-nodistsim.tagger"

let tagReader (reader:Reader) = 
    let tagger = MaxentTagger(model)
    MaxentTagger.tokenizeText(reader)
    |> Iterable.toSeq
    |> Seq.iter (fun sentence ->
            let tSentence = tagger.tagSentence(sentence :?> List)
            printfn "%O" (Sentence.listToString(tSentence, false))
        )

let tagFile (fileName:string) = 
    tagReader (new BufferedReader(new FileReader(fileName)))

let tagText (text:string) =
    tagReader (new StringReader(text))

File: StanfordSoftware/Samples/StanfordPOSTagger.Csharp.Samples/TaggerDemo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using java.io;
using java.util;

using edu.stanford.nlp.ling;
using edu.stanford.nlp.tagger.maxent;

namespace StanfordPOSTagger.Csharp.Samples
{
    public static class TaggerDemo
    {
        public const string Model =
            @"..\..\..\..\temp\stanford-postagger-2013-06-20\models\wsj-0-18-bidirectional-nodistsim.tagger";

        private static void TagReader(Reader reader)
        {
            var tagger = new MaxentTagger(Model);
            foreach (List sentence in MaxentTagger.tokenizeText(reader).toArray())
            {
                var tSentence = tagger.tagSentence(sentence);
                System.Console.WriteLine(Sentence.listToString(tSentence, false));
            }
        }

        public static void TagFile (string fileName)
        {
            TagReader(new BufferedReader(new FileReader(fileName)));
        }

        public static void TagText(string text)
        {
            TagReader(new StringReader(text));
        }
    }
}
